sharktank_models/llama3.1/test_llama.py

```diff
@@ -50,10 +64,39 @@ def compile_llama_hip(llama_mlir):
     )
 
 
+@pytest.fixture
+def compile_llama_tp2_cpu(llama_tp2_mlir):
+    return iree.compiler.compile_file(
+        llama_tp2_mlir,
+        extra_args=[
+            "--iree-hal-target-device=llvm-cpu",
+            "--iree-hal-target-device=llvm-cpu",
+            "--iree-llvmcpu-target-cpu=host",
+        ],
+    )
+
+
+@pytest.fixture
+def compile_llama_tp2_hip(llama_tp2_mlir):
+    if "HIP_TARGET" not in os.environ:
+        raise RuntimeError("HIP_TARGET not set")
+
+    target_gpu = os.environ["HIP_TARGET"]
+    return iree.compiler.compile_file(
+        llama_tp2_mlir,
+        extra_args=[
+            "--iree-hal-target-device=hip",
+            "--iree-hal-target-device=hip",
+            f"--iree-hip-target={target_gpu}",
+        ],
+    )
+
+
 class ToyLlama:
-    def __init__(self, compiled, device_id, irpa):
+    def __init__(self, compiled, device_id, irpa, sharding=1):
```

Comment:
Can you add a docstring and maybe type hints for this init function, particularly for the new `sharding` parameter? ("Number of homogeneous devices to shard the model across?")

Reply:
done.